src/main/java/com/exadel/discount/dto/DiscountDTO.java

```diff
@@ -0,0 +1,39 @@
+package com.exadel.discount.dto;
+
+import com.exadel.discount.dto.validation.Create;
+import com.exadel.discount.entity.Category;
+import lombok.Data;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Null;
+import javax.validation.constraints.Size;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Data
+public class DiscountDTO {
+
+    @Null(groups = Create.class)
+    private UUID id;
+
+    @NotNull(groups = Create.class)
+    private Category category;
+
+    @NotBlank(groups = Create.class)
+    private String name;
+    @Null(groups = Create.class)
+    private String description;
+    @NotBlank(groups = Create.class)
+    @Size(max = 50, groups = Create.class)
+    private String promo;
+    @NotNull(groups = Create.class)
+    private int percent;
+    @Null(groups = Create.class)
+    private LocalDateTime startTime;
```

Comment:
may be , Keep the ability to put start time at. creation: no annotation at field?

Reply:
+